tools/failures.py

```diff
@@ -3,269 +3,288 @@
+                         passwd="root",
+                         db="ouija")
+
+    cur = db.cursor()
+    cur.execute(query)
+
+    results = []
+    # each row is in ('val',) format, we want 'val'
+    for rows in cur.fetchall():
+        results.append(rows[0])
+
+    cur.close()
+    return results
 
-    depth_level = 2
-    depth = 0
-    results = {}
-    filtered_scenarios = [[]]
-    while depth < depth_level:
-        depth += 1
-        tempResults = {}
-        l2scenarios = copy.deepcopy(scenarios)
-        count = 0
-        for scenario in filtered_scenarios:
-            count += 1
-            temp = copy.deepcopy(failures)
-            if scenario == []:
-                filtered_scenarios = copy.deepcopy(scenarios)
-                l2scenarios = copy.deepcopy(scenarios)
-
-            add_to_scenario = []
-            for i in scenario:
-                temp, removed = removeType(temp, [i])
-                if [i] in l2scenarios:
-                    l2scenarios.remove([i])
-                    add_to_scenario.append([i])
-
-            tempResults = bruteForcePercentage(total, l2scenarios, temp, tempResults, scenario)
-
-            l2scenarios.extend(add_to_scenario)
-
-        tempResults = removeBadScenarios(tempResults, 99)
-        filtered_scenarios = buildScenarios(tempResults)
-        results = tempResults
-        print "finished round %s with %s(%s) scenarios, %s" % (depth, len(l2scenarios), len(scenarios), len(filtered_scenarios))
-        
-
-    print "\n\nsummary of choices:"
-    for index in sorted(results.iterkeys()):
-        l = sortList(results[index])
-        print "%s%% detection, %s choices- removing any of these configs:" % (index, len(l))
-        for i in l:
-            print "  * %s" % i
+def check_data(query_date):
+    data = run_query('select jobtype from seta where date="%s"' % query_date)
+    if not data:
+        print "The database does not have data for the given %s date." % query_date
+        for date in range(-3, 4):
+            current_date = query_date + datetime.timedelta(date)
+            tuple = run_query('select jobtype from seta where date="%s"' % current_date)
+            if tuple:
+                print "The data is available for date=%s" % current_date
+        return None
+    return data
+
+def print_diff(options):
+    options.end_date = datetime.datetime.strptime(options.end_date, "%Y-%m-%d")
+    options.start_date = datetime.datetime.strptime(options.start_date, "%Y-%m-%d")
+    start_tuple = check_data(options.start_date)
+    end_tuple = check_data(options.end_date)
+
+    if start_tuple is None or end_tuple is None:
+        return
+    else:
+        print "The tuples present on %s and not present on %s: " % (options.end_date, options.start_date)
+        result = list(set(start_tuple) - set(end_tuple))
+        print result
+        if len(result) > 20:
+            print "Warning: Too many differences detected, most likely there is data missing or incorrect"
```

Comment:
I would do a if else here:
if len(result) > 20:
  print warning
else:
  print result

likewise for the line below.